gen/gen_test.go

```diff
@@ -155,21 +155,75 @@ func TestStruct_ConfigurableFieldTags(t *testing.T) {
 	}
 }
 
-func TestStruct_GenFromRecordSchema(t *testing.T) {
-	schema, err := os.ReadFile("testdata/golden.avsc")
-	require.NoError(t, err)
+func TestStruct_ConfigurableLogicalTypes(t *testing.T) {
+	schema := `{
+  "type": "record",
+  "name": "test",
+  "fields": [
+    { "name": "id", "type": {"type": "string", "logicalType": "uuid"} }
+  ]
+}`
 
-	gc := gen.Config{PackageName: "Something"}
-	file, _ := generate(t, string(schema), gc)
+	gc := gen.Config{
+		PackageName: "Something",
+		LogicalTypes: []gen.LogicalType{{
+			Name:             "uuid",
+			Typ:              "uuid.UUID",
+			ThirdPartyImport: "github.com/google/uuid",
+		}},
+	}
+	_, lines := generate(t, schema, gc)
 
-	if *update {
-		err = os.WriteFile("testdata/golden.go", file, 0600)
-		require.NoError(t, err)
+	for _, expected := range []string{
+		"package something",
+		"import (",
+		"\"github.com/google/uuid\"",
+		"type Test struct {",
+		"ID uuid.UUID `avro:\"id\"`",
+		"}",
+	} {
+		assert.Contains(t, lines, expected)
+	}
+}
+
+func TestStruct_GenFromRecordSchema(t *testing.T) {
+	tests := map[string]struct {
+		config   gen.Config
+		fileName string
+	}{
+		"without custom logical types": {
+			config:   gen.Config{PackageName: "Something"},
+			fileName: "testdata/golden.go",
+		},
+		"with logical types": {
+			config: gen.Config{PackageName: "Something", LogicalTypes: []gen.LogicalType{{
+				Name:             "uuid",
+				Typ:              "uuid.UUID",
+				ThirdPartyImport: "github.com/google/uuid",
+			}}},
+			fileName: "testdata/golden_logicaltype.go",
+		},
+	}
```

Comment:
I would prefer this split into 2 tests. There are other golden tests, each in their own tests, so this does not really match.

Reply:
separated tests into their own